opm/autodiff/BlackoilWellModel_impl.hpp

```diff
@@ -281,6 +282,7 @@ namespace Opm {
             if (initial_step_) {
                 // update the explicit quantities to get the initial fluid distribution in the well correct.
                 calculateExplicitQuantities();
```

Comment:
The changes in BlackoilWellMode_impl.hpp can have some similar concept with my fix, while I need to check in details.

Reply:
Since the VPF + flow issue has been there for a while and is the last remaining part where flow_legacy has a better match then flow, I gave it spin this morning. If you already have a fix I would not spend more time on this. If you fire up a PR I could take a look at your solution.

Reply:
Basically, my fix is to get perforation density initialized before using it to do the VFP calculation. I think there are quite some good things in this PR. This PR should go in after some discussion. I will review it.

Reply:
I will suggest to put the function `calculateExplicitQuantities` at the beginning of the function `prepareTimeStep`. Then this PR can go in.